lib/rules/selector-max-specificity/index.js

```diff
@@ -77,49 +79,52 @@ const rule = (primary, secondaryOptions) => {
 		/**
 		 * Calculate the specificity of a simple selector (type, attribute, class, ID, or pseudos's own value).
 		 *
-		 * @param {string} selector
-		 * @returns {SpecificityArray}
+		 * @param {import('postcss-selector-parser').Node} node
+		 * @returns {Specificity}
 		 */
-		const simpleSpecificity = (selector) => {
-			if (optionsMatches(secondaryOptions, 'ignoreSelectors', selector)) {
+		const simpleSpecificity = (node) => {
+			if (optionsMatches(secondaryOptions, 'ignoreSelectors', node.toString())) {
 				return zeroSpecificity();
 			}
 
-			const calculatedSpecificity = specificity.calculate(selector);
-
-			assert(calculatedSpecificity[0]);
-
-			return calculatedSpecificity[0].specificityArray;
+			return selectorSpecificity(node);
 		};
 
 		/**
 		 * Calculate the the specificity of the most specific direct child.
 		 *
 		 * @param {import('postcss-selector-parser').Container<unknown>} node
-		 * @returns {SpecificityArray}
+		 * @returns {Specificity}
 		 */
 		const maxChildSpecificity = (node) =>
 			node.reduce((maxSpec, child) => {
 				const childSpecificity = nodeSpecificity(child); // eslint-disable-line no-use-before-define
 
-				return specificity.compare(childSpecificity, maxSpec) === 1 ? childSpecificity : maxSpec;
+				return compare(childSpecificity, maxSpec) > 0 ? childSpecificity : maxSpec;
 			}, zeroSpecificity());
 
 		/**
 		 * Calculate the specificity of a pseudo selector including own value and children.
 		 *
 		 * @param {import('postcss-selector-parser').Pseudo} node
-		 * @returns {SpecificityArray}
+		 * @returns {Specificity}
 		 */
 		const pseudoSpecificity = (node) => {
 			// `node.toString()` includes children which should be processed separately,
 			// so use `node.value` instead
-			const ownValue = node.value;
-			const ownSpecificity =
-				ownValue === ':not' || ownValue === ':matches'
-					? // :not and :matches don't add specificity themselves, but their children do
-					  zeroSpecificity()
-					: simpleSpecificity(ownValue);
+			const ownValue = node.value.toLowerCase();
+
+			if (ownValue === ':where') {
+				return zeroSpecificity();
+			}
+
+			let ownSpecificity;
+
+			if (optionsMatches(secondaryOptions, 'ignoreSelectors', ownValue)) {
+				ownSpecificity = zeroSpecificity();
+			} else {
+				ownSpecificity = selectorSpecificity(node.clone({ nodes: [] }));
+			}
```

Comment:
[note] We need to keep the current logic parsing each selector node AST because the `ignoreSelectors` option needs to check pseudo-class arguments.